deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -442,6 +442,22 @@ async def create_new_job(
 
         priority = 10
         async with request.state.pgpool.acquire() as con:
+            # check if pulse3d version is available
+            # if deprecated and end of life date passed then cancel the upload
+            # if end of life date is none then pulse3d version is usable
+            pulse3d_version_status = await con.fetchrow(
+                "SELECT state, end_of_life_date FROM pulse3d_versions WHERE version = $1", details.version
+            )
+            status_name = pulse3d_version_status["state"]
+            end_of_life_date = pulse3d_version_status["end_of_life_date"]
+
+            if status_name == "deprecated" and (
+                end_of_life_date is not None
```

Comment:
With the DB constraint, `end_of_life_date` should never be done for deprecated versions, so we can probably remove this check eventually, but it's fine for now